playbooks/install_stack.yaml

```diff
@@ -153,11 +153,38 @@
   - name: Install the tripleo client
     yum:
       name:
-      - ceph-ansible
       - python3-tripleoclient
     become: true
     become_user: root
 
+  - name: Prepare the deployment for Ceph
+    when: ceph_enabled
+    block:
+      - name: Check if ceph-ansible can be used to deploy Ceph
+        stat:
+          path: /usr/share/openstack-tripleo-heat-templates/environments/ceph-ansible
+        register: st_ceph
+      - name: Create ceph_ansible_enabled fact
+        set_fact:
+          ceph_ansible_enabled: "{{ st_ceph.stat.isdir is defined and st_ceph.stat.isdir | ternary(true, false) }}"
+      - name: Create ceph_env_name fact
+        set_fact:
+          ceph_env_name: "{{ ceph_ansible_enabled | ternary('ceph-ansible', 'cephadm') }}"
+      - name: Install ceph-ansible
+        yum:
+          name:
+          - ceph-ansible
+        become: true
+        become_user: root
+        when: ceph_ansible_enabled
+      - name: Install cephadm
+        yum:
+          name:
+          - cephadm
+        become: true
+        become_user: root
+        when: not ceph_ansible_enabled
```

Comment:
I don't think you need the conditionals here because you've already got `ceph_env_name`:
```yaml
- name: Install {{ ceph_env_name }}
  yum:
    name:
      {{ ceph_env_name }}
  become: true
  become_user: root
```